examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,527 @@
+
+
+###############################################################################
+# As suggested in [4]_, we can further interpret these probabilities using the
+# same criteria as the Frequentist approach: Is the probability of falling
+# inside the ROPE bigger than 95% (alpha value of 5%)?  In that case we can
+# conclude that both models are practically equivalent.
+
+
+###############################################################################
+# The Bayesian estimation approach also allows us to compute how uncertain we
+# are about our estimation of the difference. This can be calculated using
+# credible intervals which compute the range of values that have a particular
+# probability of containing the true mean of differences.
+#
+# Let's determine the credible intervals of our data using 50%, 75% and 95%:
+
+cred_intervals = []
+intervals = [0.5, 0.75, 0.95]
+
+for interval in intervals:
+    cred_interval = list(t_post.interval(interval))
+    cred_intervals.append([interval, cred_interval[0], cred_interval[1]])
+
+cred_int_df = pd.DataFrame(
+    cred_intervals,
+    columns=['interval', 'lower value', 'upper value']
+    ).set_index('interval')
+cred_int_df
+
+
+###############################################################################
+# Pairwise comparison of all models: Frequentist approach
+# -------------------------------------------------------
+#
+# We could also be interested in comparing the performance of all our models
+# evaluated with :class:`~sklearn.model_selection.GridSearchCV`. In this case
+# we would be running our statistical test multiple times, which leads us to
+# the `multiple comparisons problem
+# <https://en.wikipedia.org/wiki/Multiple_comparisons_problem>`_.
+#
+# There are many possible ways to tackle this problem, but a standard approach
+# is to apply a `Bonferroni correction
+# <https://en.wikipedia.org/wiki/Bonferroni_correction>`_. Bonferroni can be
+# computed by multiplying the p-value by the number of comparisons we are
+# testing.
+#
+# Let's compare the performance of the models using the corrected t-test:
+
+n_comparisons = (
+    factorial(len(model_scores))
+    / (factorial(2) * factorial(len(model_scores) - 2))
+    )
+pairwise_t_test = []
+
+for model_i, model_k in combinations(range(len(model_scores)), 2):
+    model_i_scores = model_scores.iloc[model_i].values
+    model_k_scores = model_scores.iloc[model_k].values
+    differences = model_i_scores - model_k_scores
+    t_stat, p_val = compute_corrected_ttest(
+        differences, n, df, n_train, n_test
+        )
+    p_val *= n_comparisons  # implement Bonferroni correction
+    if p_val > 1:
+        p_val = 1  # Bonferroni can output p-values higher than 1
+    pairwise_t_test.append(
+        [model_scores.index[model_i], model_scores.index[model_k],
+            t_stat, p_val]
+        )
+
+pairwise_comp_df = pd.DataFrame(
+    pairwise_t_test,
+    columns=['model_1', 'model_2', 't_stat', 'p_val']
+    ).round(3)
```

Comment:
Here and later. When applying black style, you will need to remove 4 spaces back with the closing parenthesis or bracket.